android/libraries/rib-base/src/main/java/com/badoo/ribs/android/ActivityStarter.kt

```diff
@@ -13,29 +13,30 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.uber.rib.core;
+package com.badoo.ribs.android
 
-import android.content.Intent;
+import android.content.Intent
+import android.os.Bundle
 
 /**
  * Start activities. A much cleaner dependency than an entire activity or context, and easier to
  * inject and mock in tests.
  */
-public interface ActivityStarter {
+interface ActivityStarter {
 
-  /**
-   * Start an activity with the given intent.
-   *
-   * @param intent The intent to open a new activity.
-   */
-  void startActivity(Intent intent);
+    /**
+     * Start an activity with the given intent.
+     *
+     * @param intent The intent to open a new activity.
+     */
+    fun startActivity(intent: Intent)
```

Comment:
How about doing startActivity(factory: Context.() -> Intent) ?
Then you don't need IntentCreator

Reply:
You will still need Context in Node to call this factory method then, and that would mean passing Context in default Rib.Dependency, which would mean it's available everywhere. I guess we should avoid that, shouldn't we?

Reply:
But the Context will be required in the implementation only

```
//Activity

fun startActivity(factory: Context.() -> Intent) {
    startActivity(factory())
}
```

```
// Rib

activityLauncher.startActivity {
    Intent(this, Activity::class.java)
}
```

Reply:
@zsoltk @ShikaSD maybe we can even do `startActivity(factory: IntentFactory.() -> Unit)`?

Reply:
Fixed as per your suggestions